modules/cms/widgets/AssetList.php

```diff
@@ -664,6 +665,8 @@ protected function checkUploadPostback()
              */
             $uploadedFile->move($this->getCurrentPath(), $uploadedFile->getClientOriginalName());
 
+            @chmod($this->getCurrentPath().'/'.$uploadedFile->getClientOriginalName(), octdec(Config::get('cms.defaultMask.file')));
```

Comment:
```suggestion
            @chmod($this->getCurrentPath() . '/' . $uploadedFile->getClientOriginalName(), octdec(Config::get('cms.defaultMask.file')));
```

Reply:
@kmarty why not just use $uploadedFile->getPath()?

Reply:
Perhaps because of my ignorance :-/.
But anyway, _$this->getCurrentPath()_ contains destination path whilst _$uploadedFile->getPath()_ contains source path (tmp path).
Just guess, but after _$uploadedFile->move()_ the file is already not in _$uploadedFile->getPath()_.

Reply:
Looked into it further and `->move()` returns the new file reference, so you can do `$uploadedFile = $uploadedFile->move()` and then `$uploadedFile->getPath()` should work

Reply:
OK, I said nothing, then :-)